refactor code, support dynamic (#8)
DevExpress-Examples/blazor-multi-tab-ui: README.md

> 

Reviewer: I have the following comments regarding the above lines:

Line 9:
"The example creates an interactive, multi-tab web interface using..."

Please change "The example" to "This example".

Line 13:
"It illustrates how end users can create personalized workspaces and multitask effectively."

Can you move this sentence to Line 9, so that it's the second (last) sentence in that paragraph? I think this will improve the reading flow.

> ### Organize Content Into Tabs

Reviewer: ```suggestion
### Organize Content into Tabs
```

> The [MdiTabs](CS/blazor_multi_tab_ui/Components/MDI/MdiTabs.razor) custom component is based on [DxTabs](https://docs.devexpress.com/Blazor/DevExpress.Blazor.DxTabs) control.

Reviewer: ```suggestion
The [MdiTabs](CS/blazor_multi_tab_ui/Components/MDI/MdiTabs.razor) custom component is based on the [DxTabs](https://docs.devexpress.com/Blazor/DevExpress.Blazor.DxTabs) control.
```

> Tabs are iteratively rendered from a persisted state collection (see details below). The content within each tab is loaded dynamically using a [DynamicComponent](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.dynamiccomponent). This approach allows for the flexible rendering of different components within the tabs without the need to hardcode them.

Reviewer: `Tabs are iteratively rendered from a persisted state collection (see details below).`

Can you clarify "iteratively" in this case? For example, can we simply say:

`Tabs are rendered from a persisted state collection (see details below).`

Or are you trying to emphasize that the process is iterative? For example:

`Tabs are rendered by iterating over a persisted state collection (see details below).`

    {
        var tabModel = tabsCollection[i];
        <DxTabPage  AllowClose="true"
                    CssClass="@TabCssClass"
                    VisibleIndex="@tabModel.VisibleIndex"
                    Visible="@tabModel.Visible"
                    Text="@tabModel.Text">
            <ChildContent>
                @if (stateService.TryGetType(tabModel.TabTypeName, out var type))
                {
                    <DynamicComponent @key="@tabModel.Id" Type="type" Parameters="tabModel.Parameters" />
                }
                else
                {
                    <DynamicComponent @key="@tabModel.Id" Type="@typeof(Unknown)" />
                }

            </ChildContent>
        </DxTabPage>
    }
</DxTabs>
```

### Persist Tab State

The state of the multi-tab interface is managed by the following classes:

- `MdiTabModel` ([MdiTabModel.cs](CS/blazor_multi_tab_ui/Models/MdiTabModel.cs)) encapsulates properties associated with each individual tab: unique identifier, visibility, title, and so on.
- `MdiStateModel` ([MdiStateModel.cs](CS/blazor_multi_tab_ui/Models/MdiStateModel.cs)) contains the list of all tabs (`MdiTabModel`) and the index of the active tab.

> - `MdiStateService` ([MdiStateService.cs](CS/blazor_multi_tab_ui/Services/MdiStateService.cs)) exposes methods for tab state management and maintains tab layout across sessions even after the user closes and reopens the browser. It serializes `MdiStateModel` to JSON and saves it to the browser's local storage every time the UI layout changes. Tab state is restored in the `OnInitializedAsync` method of the `MdiTabs` component.

Reviewer: ```suggestion
- `MdiStateService` ([MdiStateService.cs](CS/blazor_multi_tab_ui/Services/MdiStateService.cs)) exposes methods for tab state management and maintains tab layout across sessions even after the user closes and reopens the browser. It serializes `MdiStateModel` to JSON and saves it to the browser's local storage every time the UI layout changes. The tab state is restored in the `MdiTabs` component's `OnInitializedAsync` method.
```

> To ensure the state model (`MdiStateModel`) accurately reflects the live interface, implement event handlers for `TabReorder`, `TabClosing`, and `ActiveTabIndexChanged`. These handlers will listen for user actions and dynamically update the state to match the current tab layout.

Reviewer: ```suggestion
To ensure that the state model (`MdiStateModel`) accurately reflects the live interface, implement event handlers for `TabReorder`, `TabClosing`, and `ActiveTabIndexChanged`. These handlers will listen for user actions and dynamically update the state to match the current tab layout.
```

### Add Context Menu to Tabs

Implement a context menu that allows users to manage tabs as needed:

- **Close** the current tab.
- **Close** all tabs.
- **Close** all tabs except for the current one.
- **Hide** the current tab.
- **Hide** all tabs.
- **Hide** all tabs except for the current one.
- **Restore** hidden tabs.

> Create a custom [`TabsContextMenu`](CS/blazor_multi_tab_ui/Components/MDI/TabsContextMenu.razor) component which contains [DxContextMenu](https://docs.devexpress.com/Blazor/DevExpress.Blazor.DxContextMenu) and all related actions. Place it inside the `MdiTabs` component.

Reviewer: ```suggestion
Create a custom [`TabsContextMenu`](CS/blazor_multi_tab_ui/Components/MDI/TabsContextMenu.razor) component that contains [DxContextMenu](https://docs.devexpress.com/Blazor/DevExpress.Blazor.DxContextMenu) and all related actions. Place it inside the `MdiTabs` component.
```

```razor
<TabsContextMenu @ref="contextMenu" TabSelector=@($".{TabCssClass}") />
```

Use the `TabSelector` parameter to associate the specific tab with the context menu.

Implement a client-side script [`TabsContextMenu.razor.js`](CS/blazor_multi_tab_ui/Components/MDI/TabsContextMenu.razor.js) that handles right-clicks on tabs and performs the following actions:

> - Find the matching elements by their `CssClass` property.

Reviewer: ```suggestion
- Finds the matching elements by their `CssClass` property.
```

> - Suppress the default browser context menu.

Reviewer: ```suggestion
- Suppresses the default browser context menu.
```

> - Capture the mouse position, and invoke a .NET `[JSInvokable]` method that opens the context menu at the pointer's coordinates.

Reviewer: ```suggestion
- Captures the mouse position and invokes a .NET `[JSInvokable]` method that opens the context menu at the pointer's coordinates.
```
